components/services/brave_wallet/zcash/zcash_decoder.cc

```diff
@@ -0,0 +1,84 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/services/brave_wallet/zcash/zcash_decoder.h"
+
+#include <utility>
+
+#include "base/big_endian.h"
+#include "brave/components/services/brave_wallet/public/cpp/utils/protobuf_utils.h"
+#include "brave/components/services/brave_wallet/public/proto/zcash_grpc_data.pb.h"
+
+namespace brave_wallet {
+
+ZCashDecoder::ZCashDecoder() = default;
+
+ZCashDecoder::~ZCashDecoder() = default;
+
+void ZCashDecoder::ParseRawTransaction(const std::string& data,
```

Comment:
Can we perform additional data validation checks while we're parsing these structs here since it's still out of process?